app/components/Settings/Preview/TableTree.react.js

```diff
@@ -78,45 +167,18 @@ class TableTree extends Component {
     render() {
         const schemaRequest = this.props.schemaRequest || {};
         const status = schemaRequest.status;
-
-        if (typeof status === 'undefined' || status === 'loading') {
-            return (<div className="loading">{'Loading'}</div>);
-        }
-
-        if (status !== 200) {
-            return (
-                <div style={{padding: '5px', fontSize: '12px'}}>
-                    {`ERROR ${JSON.stringify(schemaRequest)}`}
-                </div>
-            );
-        }
-
         const preview = this.props.preview || {};
         const treeSchema = preview.treeSchema;
 
-        if (!treeSchema) {
-            return (<div className="loading">{'Updating'}</div>);
-        }
+        this.isConnecting(status, schemaRequest, treeSchema);
 
         const label = this.getLabel(this.props.connectionObject);
         const labelNode = <span className="node">{label}</span>;
 
         return (
             <div style={{padding: '5px 0 0 10px'}}>
                 <TreeView key={label} nodeLabel={labelNode} defaultCollapsed={false}>{
-                    Object.getOwnPropertyNames(treeSchema).sort().map(tableName => {
```

Comment:
Similarly, here. Please, don't move this code into `getTreeNodes`. `render` is so small enough that it's easier to  read it in one go.